client/core/core.go

```diff
@@ -1306,6 +1306,11 @@ func (c *Core) Trade(pw []byte, form *TradeForm) (*Order, error) {
 		return nil, fmt.Errorf("unknown DEX %s", form.DEX)
 	}
 
+	// Cannot trade before logging in.
+	if !dc.acct.authed() {
+		return nil, errors.New("cannot trade: account not authed")
+	}
+
```

Comment:
For the rpcclient, you can try to trade before logging in. Really, you must have logged in for dc.acct.privKey to be populated, but this stops that from being possible. Is it reasonable to require the account be authed?

Reply:
I changed this to erroring on sign when the account is locked